.github/workflows/engine-and-cli.yml

```diff
@@ -1,86 +0,0 @@
-name: Engine & CLI
-
-on:
-  push:
-    branches: ["main"]
-  pull_request:
-    types:
-      - opened
-      - synchronize
-      - reopened
-      - ready_for_review
-  # Enable manual trigger for easy debugging
-  workflow_dispatch:
-
-permissions:
-  contents: read
-  pull-requests: write
-
-concurrency:
-  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
-  cancel-in-progress: true
-
-jobs:
-  lint:
```

Comment:
Also, I think there's an issue where actually the current split doesn't actually cover all of the test suites? For example, `TestLegacy` which is actually one of our more complex test suites, since it tests for multiple API versions through the dagger API to ensure compatibility.

Just noticed this now (sorry! this is really inconvenient to notice only right now). We should definitely be ensuring that our split actually covers everything (part of the reason we should do so in code). But potentially, does this raise issues about the results we've been recording? If not all test suites have been part of our numbers, then we might not be able to trust the metrics we've gathered so far (anecdotally, I know a good number of our flakes have come from `TestLegacy`, or at least used to, a couple months ago).

Reply:
Also, it appears there's no `testdev-everything-else` in the split?

So test suites like `TestLegacy` don't seem to be running under the `testdev` setup? If so, could the difference in the numbers between both the split and the non-split be explained by the fact that some groups of tests just aren't run in the split in testdev?

(Note: wrote an earlier version of this comment which wasn't correct, so removed it and wrote this one)